content/docs/develop/userscripts/about-userscripts.md

```diff
@@ -0,0 +1,136 @@
+---
+title: About userscripts
+description: Userscripts are JavaScript files that are executed every time the user loads a Scratch page. They can modify the document's HTML, add new buttons, customize Scratch editor behavior, and so much more.
+aliases: 
+  - /docs/developing/about-userscripts
+---
+
+Userscripts are JavaScript files that are executed every time the user loads a Scratch page. They can modify the document's HTML, add new buttons, customize Scratch editor behavior, and so much more.
+
+Similarly to userscripts that you might download for userscript managers like Tampermonkey or Greasemonkey, Scratch Addons userscripts consist of pieces of JavaScript that are executed in the same execution context as the JavaScript code from Scratch itself. In browser extension vocabulary, this execution context is often called the "main world".
+
+Even though Scratch Addons userscripts are part of a browser extension, they cannot access any `chrome.*` or `browser.*` APIs. Instead, Scratch Addons offers an [`addon.*` API](/docs/reference/addon-api/). 
+
+
+## Declaring userscripts in the addon manifest
+
+{{< admonition warning >}}
+**Some changes require an extension reload** from `chrome://extensions` to take effect, such as updating the addon manifest file.
+
+It's not necessary to reload the extension when changing the source of an already existing userscript JavaScript file. In those cases, reloading the page is enough.
+{{< /admonition >}}
+
+Userscripts are declared inside a "userscripts" array.
+
+Each item of the array must have the following properties:
+- `"url"`: the relative URL to a JavaScript file.
+- `"matches"`: the list of Scratch pages where the userscript will run. See [matches](/docs/reference/addon-manifest/#matches) for more information.
+
+Example manifest:
+```json
+{
+  "name": "Scratch Messaging",
+  "description": "Provides easy reading and replying to your Scratch messages.",
+  "userscripts": [
+    {
+      "url": "userscript.js",
+      "matches": ["projects", "https://scratch.mit.edu/", "profiles"]
+    }
+  ],
+  "tags": ["community"],
+  "enabledByDefault": false
+}
+```
```

Comment:
I saw [`versionAdded` is required](https://scratchaddons.com/docs/reference/addon-manifest/#versionadded). Would it better to include it?

Reply:
And I'll have to double-check whether `enabledByDefault` is actually required.

Reply:
`enabledByDefault` has a default of `false`, so I don't think it is required.